API.md

```diff
@@ -0,0 +1,440 @@
+```
+
+This set of methods sets the location the map is centered on, the zoom level,
+the heading and the pitch (Android only) of the map.
+
+The transition to the desired location is animated by default, but can be made
+instantaneous by passing `animated` as `false`.
+
+For `easeTo`, all arguments inside the options object are optional. You can specify
+any combination of center coords, zoomLevel, direction and pitch. What is not
+specified stays at their current values.
+
+The methods accept an optional `callback` that will get fired when the animation
+has ended. Additionally, the return value is a promise that gets resolved when the
+animation has ended.
+
+---
+
+```javascript
+this._map.setVisibleCoordinateBounds(latitudeSW, longitudeSW, latitudeNE, longitudeNE, paddingTop = 0, paddingRight = 0, paddingBottom = 0, paddingLeft = 0, animated = true);
+```
+
+This method adjusts the center location and the zoomLevel of the map so that
+the rectangle determined by `latitudeSW`, `longitudeSW`, `latitudeNE`,
+`longitudeNE` fits inside the viewport.
+
+You can optionally pass a minimum padding (in screen points) that will be
+visible around the given coordinate bounds.
+
+The transition is animated unless you pass `animated` as `false`.
+
+---
+
+```javascript
+this._map.setCamera(latitude, longitude, fromDistance, pitch, direction, animationDuration = 0.3); // iOS Only
+```
+
+This method only works on iOS.
+
+Sets the map pitched at an angle (`pitch`, measured in
+degrees, 0 is from straight above), looking towards `latitude` and `longitude`
+from `fromDistance` meters away, pointing towards heading `direction`.
+
+Use `animationDuration` to adjust the speed of the transition.
+
+---
+
+```javascript
+this._map.getCenterCoordinateZoomLevel(data => {
+  // ...
+});
+```
+
+Gets the current coordinates and zoom level of the map.
+
+`data` is an object of the form `{ latitude, longitude, zoomLevel }`
+
+---
+
+```javascript
+this._map.getDirection(direction => {
+  // ...
+});
+```
+
+Gets the current heading of the map.
+
+`direction` is the heading in degrees.
+
+---
+
+```javascript
+this._map.getPitch(pitch => { // Android only
+  // ...
+});
+```
+
+Gets the current tilt of the map. (Android only)
+
+`pitch` is the tilt in degrees measured from the normal to the map.
+
+---
+
+```javascript
+this._map.getBounds(bounds => {
+  // ...
+});
+```
+
+Gets the bounding rectangle in GPS coordinates that is currently visible on
+within the map's viewport.
+
+`bounds` is an array representing `[ latitudeSW, longitudeSW, latitudeNE, longitudeNE ]`
+
+---
+
+```javascript
+this._map.selectAnnotation(id, animated = true);
+```
+
+Selects the annotation tagged with `id`, as if it would be tapped by the user.
+
+The transition is animated unless you pass `animated` as `false`.
+
+## Styles
+
+#### Default styles
+
+Mapbox GL ships with 6 included styles:
+
+* `Mapbox.mapStyles.streets`
+* `Mapbox.mapStyles.emerald`
```

Comment:
Nice to know. I just shipped an app with emerald. I can add a deprecation warning to it.